checker/src/org/checkerframework/checker/index/lowerbound/LowerBoundAnnotatedTypeFactory.java

```diff
@@ -300,6 +301,25 @@ public Void visitMemberSelect(MemberSelectTree tree, AnnotatedTypeMirror type) {
         }
 
         /**
+         * Compound assignments aren't desugared correctly in some way, which prevents x from
+         * being @Positive after, e.g., this code is executed: "x = 5; x -= 2;". This code is more
+         * general and ensures that any identifier with a compile-time known value is represented
+         * correctly.
+         */
+        @Override
+        public Void visitIdentifier(IdentifierTree tree, AnnotatedTypeMirror type) {
```

Comment:
This won't work if the type of the identifier is computed using an element instead of a tree. 

Instead do something like this:
https://github.com/typetools/checker-framework/blob/master/checker/src/org/checkerframework/checker/index/minlen/MinLenAnnotatedTypeFactory.java#L239-L259